subsys/net/ip/ipv6.c

```diff
@@ -779,13 +779,15 @@ int net_ipv6_finalize_raw(struct net_pkt *pkt, u8_t next_header)
 	NET_IPV6_HDR(pkt)->len[1] = total_len & 0xff;
 
 #if defined(CONFIG_NET_UDP)
-	if (next_header == IPPROTO_UDP) {
+	if (next_header == IPPROTO_UDP &&
+	    net_if_need_calc_tx_checksum(net_pkt_iface(pkt))) {
```

Comment:
This is kind of a hot path -- calling an indirect function for every outgoing packet, specially if it's not something that's going to change at runtime, seems a bit of overkill

Reply:
True, I will refactor this.

Reply:
After some though, I am not sure it warrants to try to optimize this too much. The reason being that we can have multiple network interface in the system with different checksum offloading configuration. This information, whether checksum offloading is enabled or not for some specific interface, would then need to be stored to the interface. So only optimization would be that instead of calling the network device driver for offloading status, we would check the offloading status from network interface.